src/QsCompiler/LanguageServer/Utils.cs

```diff
@@ -33,6 +36,23 @@ internal static void ShowInWindow(this QsLanguageServer server, string text, Mes
             _ = server.NotifyClientAsync(Methods.WindowShowMessageName, message);
         }
 
+        /// <summary>
+        /// Shows a dialog window with options (actions) to the user, and returns the selected option (action).
+        /// </summary>
+        internal static async Task<MessageActionItem> ShowDialogInWindowAsync(this QsLanguageServer server, string text, MessageType severity, MessageActionItem[] actionItems)
+        {
+            var message =
+                new ShowMessageRequestParams()
+                {
+                    Message = text,
+                    MessageType = severity,
+                    Actions = actionItems
+                };
+            QsCompilerError.Verify(server != null && message != null, "cannot show message - given server or text was null");
```

Comment:
This can't happen unless someone overrides the nullable type checks, and in that case the line below will throw NRE anyway.

```suggestion
```